src/analysis/normed_space/units.lean

```diff
@@ -41,29 +41,31 @@ def one_sub (t : α) (h : ∥t∥ < 1) : units α :=
 
 /-- In a complete normed ring, a perturbation of a unit `x` by an element `t` of distance less than
 `∥x⁻¹∥⁻¹` from `x` is a unit.  Here we construct its `units` structure. -/
-def add [nonzero α] (x : units α) (t : α) (h : ∥t∥ < ∥((x⁻¹:units α):α)∥⁻¹) : units α :=
+def add (x : units α) (t : α) (h : ∥t∥ < ∥((x⁻¹:units α):α)∥⁻¹) : units α :=
 x * (units.one_sub (- (((x⁻¹:units α):α) * t))
 begin
-  have hpos : 0 < ∥((x⁻¹:units α):α)∥ := units.norm_pos x⁻¹,
-  calc ∥-(((x⁻¹:units α):α) * t)∥
-      = ∥((x⁻¹:units α):α) * t∥                    : by { rw norm_neg }
-  ... ≤ ∥((x⁻¹:units α):α)∥ * ∥t∥                   : norm_mul_le x.inv _
-  ... < ∥((x⁻¹:units α):α)∥ * ∥((x⁻¹:units α):α)∥⁻¹ : by nlinarith only [h, hpos]
-  ... = 1                                         : mul_inv_cancel (ne_of_gt hpos),
+  rcases subsingleton_or_nonzero α with _i|_i, resetI,
```

Comment:
```suggestion
  rcases subsingleton_or_nonzero α with _i|_i; resetI,
```
if you use `;` instead of `,` here, then the `resetI` will apply to both branches, and hopefully you should be able to remove the `@` in the proof of `hpos` below (untested).